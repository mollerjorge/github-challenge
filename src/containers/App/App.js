import React, { Component } from "react";
import { connect } from "react-redux";
import { Debounce } from "react-throttle";

// Actions
import {
  getProjects,
  setProject,
  getContributors,
  setSearchValue
} from "../../redux/FProjects/actions";

// Components
import FProject from "../../components/FProject/FProject";
import FProjectsList from "../../components/FProjectsList/FProjectsList";
import FetchList from "../FetchList/FetchList";

// UI Components
import AppHolder from "./App.style.js";
import { Layout, Input, Icon } from "antd";
const { Sider, Content } = Layout;

class App extends Component {
  onClickProjectHandler = (id, name, events) => {
    this.props.setProject(this.props.projects.filter(p => p.id === id)[0]);
    this.props.getContributors(name);
  };

  onChangeSearchInput = ({ target: { value } }) => {
    this.props.setSearchValue(value);
  };

  render() {
    const {
      getProjects,
      searchValue,
      onClickProjectHandler,
      currentProject,
      contributors
    } = this.props;

    let content = "";
    if (currentProject && contributors) {
      content = (
        <FProject contributors={contributors} project={currentProject} />
      );
    } else {
      content = (
        <div
          style={{
            position: "absolute",
            transform: "translate(-50%,-50%)",
            top: "50%",
            left: "50%"
          }}
        >
          Please select a project from the sidebar
        </div>
      );
    }
    return (
      <AppHolder>
        <Layout style={{ height: "100vh", overflow: "scroll" }}>
          <Sider
            width="20rem"
            height="100vh"
            style={{ overflow: "scroll", background: "#262a34" }}
          >
            <Debounce time="500" handler="onChange">
              <Input
                placeholder="Search Projects"
                onChange={this.onChangeSearchInput}
                suffix={<Icon type="search" />}
                size="large"
                style={{ margin: "1rem", width: "18rem" }}
              />
            </Debounce>
            <FetchList
              getList={getProjects}
              render={props => {
                return (
                  <FProjectsList
                    searchValue={searchValue}
                    onClickProjectHandler={this.onClickProjectHandler}
                    projects={props.projects}
                  />
                );
              }}
            />
          </Sider>

          <Layout>
            <Content style={{ padding: "5rem", position: "relative" }}>{content}</Content>
          </Layout>
        </Layout>
      </AppHolder>
    );
  }
}

const mapStateToProps = ({
  ProjectReducer: { projects, currentProject, contributors, searchValue }
}) => {
  return {
    projects: projects,
    currentProject: currentProject,
    contributors: contributors,
    searchValue: searchValue
  };
};

export default connect(mapStateToProps, {
  getProjects,
  setProject,
  getContributors,
  setSearchValue
})(App);
