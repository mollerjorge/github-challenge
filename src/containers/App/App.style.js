import styled from 'styled-components';

const AppHolder = styled.div`

	*,
	body,
	*::after,
	*::before {
	  margin: 0;
	  padding: 0;
	  box-sizing: border-box;
	}

	min-height: 100vh;
	display: flex;
	justify-content: space-between;
	
`;

export default AppHolder;