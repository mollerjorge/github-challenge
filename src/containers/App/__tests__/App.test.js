import React from 'react';
import { shallow } from 'enzyme';
import {App} from '../App';

describe('App', () => {
	const mockFn = jest.fn();
	const mocksetSearchValue = jest.fn();
	const app = shallow(<App getProjects={mockFn} setSearchValue={mocksetSearchValue} searchValue=""/>);

	it('renders correctly', () => {
		expect(app).toMatchSnapshot();
	});

	describe('there are no projects selected', () => {

		it('it renders the `Please select a project from sidebar`', () => {
			expect(app.find('.content-text').text()).toEqual('Please select a project from the sidebar');
		});
	});

	describe('user enter text in search input', () => {

		beforeEach(() => {
			app.find('Input').dive().find('input').simulate('change', {target: {value: 'flow'} });
		});

		it('executes `setSearchValue` callback', () => {
			expect(mocksetSearchValue).toHaveBeenCalledTimes(1);
			expect(mocksetSearchValue).toHaveBeenCalledWith('flow');
		});
	})


});