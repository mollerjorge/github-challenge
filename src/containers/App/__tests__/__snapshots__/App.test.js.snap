// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    getProjects={[Function]}
    searchValue=""
    setSearchValue={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Adapter
        style={
                Object {
                        "height": "100vh",
                        "overflow": "scroll",
                      }
        }
>
        <Sider
                collapsedWidth={80}
                collapsible={false}
                defaultCollapsed={false}
                height="100vh"
                prefixCls="ant-layout-sider"
                reverseArrow={false}
                style={
                        Object {
                                "background": "#262a34",
                                "overflow": "scroll",
                              }
                }
                theme="dark"
                width="20rem"
        >
                <Debounce
                        handler="onChange"
                        handlers={Array []}
                        time="500"
                >
                        <Input
                                disabled={false}
                                onChange={[Function]}
                                placeholder="Search Projects"
                                prefixCls="ant-input"
                                size="large"
                                style={
                                        Object {
                                                "margin": "1rem",
                                                "width": "18rem",
                                              }
                                }
                                suffix={
                                        <Icon
                                                type="search"
                                        />
                                }
                                type="text"
                        />
                </Debounce>
                <Connect(FetchList)
                        getList={[Function]}
                        render={[Function]}
                />
        </Sider>
        <Adapter>
                <Adapter
                        style={
                                Object {
                                        "padding": "5rem",
                                        "position": "relative",
                                      }
                        }
                >
                        <div
                                className="content-text"
                                style={
                                        Object {
                                                "left": "50%",
                                                "position": "absolute",
                                                "top": "50%",
                                                "transform": "translate(-50%,-50%)",
                                              }
                                }
                        >
                                Please select a project from the sidebar
                        </div>
                </Adapter>
        </Adapter>
</Adapter>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Sider
            collapsedWidth={80}
            collapsible={false}
            defaultCollapsed={false}
            height="100vh"
            prefixCls="ant-layout-sider"
            reverseArrow={false}
            style={
                        Object {
                                    "background": "#262a34",
                                    "overflow": "scroll",
                                  }
            }
            theme="dark"
            width="20rem"
>
            <Debounce
                        handler="onChange"
                        handlers={Array []}
                        time="500"
            >
                        <Input
                                    disabled={false}
                                    onChange={[Function]}
                                    placeholder="Search Projects"
                                    prefixCls="ant-input"
                                    size="large"
                                    style={
                                                Object {
                                                            "margin": "1rem",
                                                            "width": "18rem",
                                                          }
                                    }
                                    suffix={
                                                <Icon
                                                            type="search"
                                                />
                                    }
                                    type="text"
                        />
            </Debounce>
            <Connect(FetchList)
                        getList={[Function]}
                        render={[Function]}
            />
</Sider>,
          <Adapter>
            <Adapter
                        style={
                                    Object {
                                                "padding": "5rem",
                                                "position": "relative",
                                              }
                        }
            >
                        <div
                                    className="content-text"
                                    style={
                                                Object {
                                                            "left": "50%",
                                                            "position": "absolute",
                                                            "top": "50%",
                                                            "transform": "translate(-50%,-50%)",
                                                          }
                                    }
                        >
                                    Please select a project from the sidebar
                        </div>
            </Adapter>
</Adapter>,
        ],
        "style": Object {
          "height": "100vh",
          "overflow": "scroll",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Debounce
                handler="onChange"
                handlers={Array []}
                time="500"
>
                <Input
                                disabled={false}
                                onChange={[Function]}
                                placeholder="Search Projects"
                                prefixCls="ant-input"
                                size="large"
                                style={
                                                Object {
                                                                "margin": "1rem",
                                                                "width": "18rem",
                                                              }
                                }
                                suffix={
                                                <Icon
                                                                type="search"
                                                />
                                }
                                type="text"
                />
</Debounce>,
              <Connect(FetchList)
                getList={[Function]}
                render={[Function]}
/>,
            ],
            "collapsedWidth": 80,
            "collapsible": false,
            "defaultCollapsed": false,
            "height": "100vh",
            "prefixCls": "ant-layout-sider",
            "reverseArrow": false,
            "style": Object {
              "background": "#262a34",
              "overflow": "scroll",
            },
            "theme": "dark",
            "width": "20rem",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Input
                  disabled={false}
                  onChange={[Function]}
                  placeholder="Search Projects"
                  prefixCls="ant-input"
                  size="large"
                  style={
                                    Object {
                                                      "margin": "1rem",
                                                      "width": "18rem",
                                                    }
                  }
                  suffix={
                                    <Icon
                                                      type="search"
                                    />
                  }
                  type="text"
/>,
                "handler": "onChange",
                "handlers": Array [],
                "time": "500",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "onChange": [Function],
                  "placeholder": "Search Projects",
                  "prefixCls": "ant-input",
                  "size": "large",
                  "style": Object {
                    "margin": "1rem",
                    "width": "18rem",
                  },
                  "suffix": <Icon
                    type="search"
/>,
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "getList": [Function],
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Adapter
              style={
                            Object {
                                          "padding": "5rem",
                                          "position": "relative",
                                        }
              }
>
              <div
                            className="content-text"
                            style={
                                          Object {
                                                        "left": "50%",
                                                        "position": "absolute",
                                                        "top": "50%",
                                                        "transform": "translate(-50%,-50%)",
                                                      }
                            }
              >
                            Please select a project from the sidebar
              </div>
</Adapter>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="content-text"
                style={
                                Object {
                                                "left": "50%",
                                                "position": "absolute",
                                                "top": "50%",
                                                "transform": "translate(-50%,-50%)",
                                              }
                }
>
                Please select a project from the sidebar
</div>,
              "style": Object {
                "padding": "5rem",
                "position": "relative",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Please select a project from the sidebar",
                "className": "content-text",
                "style": Object {
                  "left": "50%",
                  "position": "absolute",
                  "top": "50%",
                  "transform": "translate(-50%,-50%)",
                },
              },
              "ref": null,
              "rendered": "Please select a project from the sidebar",
              "type": "div",
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Adapter
          style={
                    Object {
                              "height": "100vh",
                              "overflow": "scroll",
                            }
          }
>
          <Sider
                    collapsedWidth={80}
                    collapsible={false}
                    defaultCollapsed={false}
                    height="100vh"
                    prefixCls="ant-layout-sider"
                    reverseArrow={false}
                    style={
                              Object {
                                        "background": "#262a34",
                                        "overflow": "scroll",
                                      }
                    }
                    theme="dark"
                    width="20rem"
          >
                    <Debounce
                              handler="onChange"
                              handlers={Array []}
                              time="500"
                    >
                              <Input
                                        disabled={false}
                                        onChange={[Function]}
                                        placeholder="Search Projects"
                                        prefixCls="ant-input"
                                        size="large"
                                        style={
                                                  Object {
                                                            "margin": "1rem",
                                                            "width": "18rem",
                                                          }
                                        }
                                        suffix={
                                                  <Icon
                                                            type="search"
                                                  />
                                        }
                                        type="text"
                              />
                    </Debounce>
                    <Connect(FetchList)
                              getList={[Function]}
                              render={[Function]}
                    />
          </Sider>
          <Adapter>
                    <Adapter
                              style={
                                        Object {
                                                  "padding": "5rem",
                                                  "position": "relative",
                                                }
                              }
                    >
                              <div
                                        className="content-text"
                                        style={
                                                  Object {
                                                            "left": "50%",
                                                            "position": "absolute",
                                                            "top": "50%",
                                                            "transform": "translate(-50%,-50%)",
                                                          }
                                        }
                              >
                                        Please select a project from the sidebar
                              </div>
                    </Adapter>
          </Adapter>
</Adapter>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Sider
              collapsedWidth={80}
              collapsible={false}
              defaultCollapsed={false}
              height="100vh"
              prefixCls="ant-layout-sider"
              reverseArrow={false}
              style={
                            Object {
                                          "background": "#262a34",
                                          "overflow": "scroll",
                                        }
              }
              theme="dark"
              width="20rem"
>
              <Debounce
                            handler="onChange"
                            handlers={Array []}
                            time="500"
              >
                            <Input
                                          disabled={false}
                                          onChange={[Function]}
                                          placeholder="Search Projects"
                                          prefixCls="ant-input"
                                          size="large"
                                          style={
                                                        Object {
                                                                      "margin": "1rem",
                                                                      "width": "18rem",
                                                                    }
                                          }
                                          suffix={
                                                        <Icon
                                                                      type="search"
                                                        />
                                          }
                                          type="text"
                            />
              </Debounce>
              <Connect(FetchList)
                            getList={[Function]}
                            render={[Function]}
              />
</Sider>,
            <Adapter>
              <Adapter
                            style={
                                          Object {
                                                        "padding": "5rem",
                                                        "position": "relative",
                                                      }
                            }
              >
                            <div
                                          className="content-text"
                                          style={
                                                        Object {
                                                                      "left": "50%",
                                                                      "position": "absolute",
                                                                      "top": "50%",
                                                                      "transform": "translate(-50%,-50%)",
                                                                    }
                                          }
                            >
                                          Please select a project from the sidebar
                            </div>
              </Adapter>
</Adapter>,
          ],
          "style": Object {
            "height": "100vh",
            "overflow": "scroll",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Debounce
                  handler="onChange"
                  handlers={Array []}
                  time="500"
>
                  <Input
                                    disabled={false}
                                    onChange={[Function]}
                                    placeholder="Search Projects"
                                    prefixCls="ant-input"
                                    size="large"
                                    style={
                                                      Object {
                                                                        "margin": "1rem",
                                                                        "width": "18rem",
                                                                      }
                                    }
                                    suffix={
                                                      <Icon
                                                                        type="search"
                                                      />
                                    }
                                    type="text"
                  />
</Debounce>,
                <Connect(FetchList)
                  getList={[Function]}
                  render={[Function]}
/>,
              ],
              "collapsedWidth": 80,
              "collapsible": false,
              "defaultCollapsed": false,
              "height": "100vh",
              "prefixCls": "ant-layout-sider",
              "reverseArrow": false,
              "style": Object {
                "background": "#262a34",
                "overflow": "scroll",
              },
              "theme": "dark",
              "width": "20rem",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Input
                    disabled={false}
                    onChange={[Function]}
                    placeholder="Search Projects"
                    prefixCls="ant-input"
                    size="large"
                    style={
                                        Object {
                                                            "margin": "1rem",
                                                            "width": "18rem",
                                                          }
                    }
                    suffix={
                                        <Icon
                                                            type="search"
                                        />
                    }
                    type="text"
/>,
                  "handler": "onChange",
                  "handlers": Array [],
                  "time": "500",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "disabled": false,
                    "onChange": [Function],
                    "placeholder": "Search Projects",
                    "prefixCls": "ant-input",
                    "size": "large",
                    "style": Object {
                      "margin": "1rem",
                      "width": "18rem",
                    },
                    "suffix": <Icon
                      type="search"
/>,
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "getList": [Function],
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Adapter
                style={
                                Object {
                                                "padding": "5rem",
                                                "position": "relative",
                                              }
                }
>
                <div
                                className="content-text"
                                style={
                                                Object {
                                                                "left": "50%",
                                                                "position": "absolute",
                                                                "top": "50%",
                                                                "transform": "translate(-50%,-50%)",
                                                              }
                                }
                >
                                Please select a project from the sidebar
                </div>
</Adapter>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="content-text"
                  style={
                                    Object {
                                                      "left": "50%",
                                                      "position": "absolute",
                                                      "top": "50%",
                                                      "transform": "translate(-50%,-50%)",
                                                    }
                  }
>
                  Please select a project from the sidebar
</div>,
                "style": Object {
                  "padding": "5rem",
                  "position": "relative",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Please select a project from the sidebar",
                  "className": "content-text",
                  "style": Object {
                    "left": "50%",
                    "position": "absolute",
                    "top": "50%",
                    "transform": "translate(-50%,-50%)",
                  },
                },
                "ref": null,
                "rendered": "Please select a project from the sidebar",
                "type": "div",
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
